Assets/AWSIM/Scripts/NPCs/Pedestrians/NPCPedestrian.cs

```diff
@@ -31,6 +31,25 @@ public class NPCPedestrian : MonoBehaviour
         private const string moveSpeedProperty = "moveSpeed";
         private const string rotateSpeedProperty = "rotateSpeed";
 
+        private RGLUnityPlugin.SceneManager sceneManager;
+
+        void Start()
+        {
+            sceneManager = FindObjectOfType<RGLUnityPlugin.SceneManager>();
```

Comment:
I don't think each NPC class should reference the RGL SceneManager. Is there a problem if the call to SceneManager.RegisterRGL is called by the class that spawns the NPC?

Reply:
@mackierx111 CC @pijaro 

I apologize for the late response. I see one main issue with this approach, which is that for what I know, `NPCPedestrian.cs` does not have a spawning class when AWSIM is run without SSv2. For example in Nishinjuku scene, the pedestrian GameObjects are already present in the scene.

![image](https://github.com/tier4/AWSIM/assets/2529619/657e1968-e3b4-44b6-84e7-951ec07e0f37)

This of course can be resolved by having an additional script in the scene, which is given references to all pedestrian GameObjects and is responsible for registering them with RGL. Do you want us to implement such a solution?

Reply:
> I apologize for the late response.

no problem!

In any case, I don't think we should add code that depends on other classes. If such additions are made in the future, there will be more and more code to add caches other than RGL to each component class, such as NPCs, which we do not want.

For example, the following workarounds can be suggested
- Have NPC Pedestrian spawn at the start of Simulation. (And the spawning class will do the RGL cache)
- Refer to the first placed NPC Pedestrian and add it to the RGL cache.

Reply:
Understood, we will implement changes not to have references to RGL within pedestrian classes.